src/builder.rs

```diff
@@ -302,6 +302,25 @@ impl Builder {
         Ok(())
     }
 
+    /// The net value of the bundle to be built. The value of all spends,
+    /// minus the value of all outputs.
+    ///
+    /// Useful for balancing a transaction, as the value balance of an individual bundle
+    /// can be non-zero, but a transaction may not have a positive total value balance.  
+    pub fn value_balance(&self) -> Result<ValueSum, OverflowError> {
+        (self
+            .spends
+            .iter()
+            .map(|spend| ValueSum::from_raw(spend.note.value().inner() as i64))
+            .sum::<Result<ValueSum, OverflowError>>()?
+            - self
+                .recipients
+                .iter()
+                .map(|recipient| ValueSum::from_raw(recipient.value.inner() as i64))
+                .sum::<Result<ValueSum, OverflowError>>()?)
+        .ok_or(OverflowError)
+    }
```

Comment:
I was in the middle of writing this comment when the PR was updated:

> This sum must not be computed with `i64` values, because a value sum within a bundle is a signed 64-bit integer, which is not the same as an `i64` (which is a signed _63-bit_ integer). The `as i64` casts can cause data loss resulting in financial errors, and overflows are not handled. This should use the existing types like the builder itself does, and its return value should be user-specified like it is for `Bundle`:

My suggestion still applies:
```suggestion
    pub fn value_balance<V: TryFrom<i64>>(&self) -> Result<V, value::OverflowError> {
        let value_balance = self
            .spends
            .iter()
            .map(|spend| spend.note.value() - NoteValue::zero())
            .chain(
                self.recipients
                    .iter()
                    .map(|recipient| NoteValue::zero() - recipient.value),
            )
            .fold(Some(ValueSum::zero()), |acc, note_value| acc? + note_value)
            .ok_or(OverflowError)?;

        i64::try_from(value_balance).and_then(|i| V::try_from(i).map_err(|_| value::OverflowError))
    }
```